### PR TITLE
keyless circuit

### PR DESCRIPTION
**Pull request by zjma**
## Description

Previous attempt #12839 had a bad merge. Restarting in a new PR.

- keyless circuit in `keyless/circuit-data`;
- circuit unit tests with `cargo test -p aptos-keyless-circuit-tests`
- A daily circuit test run in CI.

TBD in new PRs:
- Update command examples in `keyless/circuit-data/README.md`.

## Type of Change
- [x] New feature

## How Has This Been Tested?
New UTs

## Key Areas to Review
The circuit has been reviewed separately. This PR is just for landing it on main.

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: keyless/circuit-data/install-deps.sh [hunk 1]
```diff
@@ -51,8 +51,15 @@ install_npm_deps() {
     echo -n "snarkjs installation finished."
 }
 
+install_circomlib() {
+    echo -n "circomlib@2.0.5 installation started."
+    npm install -g circomlib@2.0.5
+    echo -n "circomlib@2.0.5 installation finished."
+}
```
Comment (zjma) [inline keyless/circuit-data/install-deps.sh:58]: bookmark 0

File: keyless/circuit-tests/src/lib.rs [hunk 2]
```diff
@@ -51,11 +50,15 @@ impl TestCircuitHandle {
         let include_root_dir = cargo_manifest_dir.join("../circuit-data/templates");
         let tmp_circuit_path = dir.path().to_owned().join("circuit.circom");
         let mut tmp_circuit_file = File::create(&tmp_circuit_path)?;
+        let global_node_modules_path =
+            String::from_utf8(Command::new("npm").args(["root", "-g"]).output()?.stdout).unwrap();
```
Comment (zjma) [inline keyless/circuit-tests/src/lib.rs:54]: bookmark 2

**Review by alinush — COMMENTED** [review]
Let's not call the circuit code directory `circuit-data`.

Also not a big fan of `common` as a name. Is there something more descriptive?

File: keyless/circuit-data/install-deps.sh
Comment (alinush) [inline keyless/circuit-data/install-deps.sh:1]: Should this be moved to be part of `./scripts/dev_setup.sh`?

File: keyless/circuit-data/templates/rsa_test.circom
Comment (alinush) [inline keyless/circuit-data/templates/rsa_test.circom:1]: Should this be moved to `tests/`?

File: keyless/circuit-data/input_gen.py
Comment (alinush) [inline keyless/circuit-data/input_gen.py:1]: I presume we will be deprecating this soon, now that we have proof simulation?

File: keyless/circuit-data/install-deps.sh
Comment (zjma) [reply keyless/circuit-data/install-deps.sh:1]: it doesn't have to. also `dev_setup.sh` is already 1k-lines script and probably needs some break-up...

File: keyless/circuit-data/templates/rsa_test.circom
Comment (zjma) [reply keyless/circuit-data/templates/rsa_test.circom:1]: dead code, deleted.

**Review by heliuchuan — APPROVED** [review]
just stamping

**Review by rex1fernando — APPROVED** [review]

**Review by alinush — APPROVED** [review]
🖖

**Pull request merged**